### PR TITLE
Update star database

### PR DESCRIPTION
**Pull request by SevenSpheres**
Credit to @ajtribick 

As mentioned in [this post](https://celestia.space/forum/viewtopic.php?p=147721#p147721), the README file needs to be updated as well, but for some reason it isn't in this repository (see #4).

### CODE REVIEW

**Comment by ajtribick** [comment]
You should definitely ensure that you have the credits issue resolved before you merge this. The README.md file in my repository includes a list of source publications which should also be preserved.

**Comment by SevenSpheres** [comment]
> please address Andrew's comment

Here or in the main repo? You're confusing me with your:
> readme should be updated in celestiacontent instead, could you copy it there?

**Comment by 375gnu** [comment]
> > 
> 
> Here or in the main repo? You're confusing me with your:
> 
> > readme should be updated in celestiacontent instead, could you copy it there?

here. i was a bit sleepy when reviewed your changes, so decided that these PRs are to the code repo :)

**Comment by SevenSpheres** [comment]
So. Where do I put the license/credit information? Should I move the data-related parts of the readme from Celestia to CelestiaContent and edit it here? I really don't want to have two readme files though.

**Comment by LukeCEL** [comment]
In keeping with the standard of including the scripts that generate data files, we also include @ajtribick's [celestia-gaia-stardb](https://github.com/ajtribick/celestia-gaia-stardb) repo in src/tools?

**Comment by SevenSpheres** [comment]
> In keeping with the standard of including the scripts that generate data files, we also include @ajtribick's celestia-gaia-stardb repo in src/tools?

Seems like a good idea.

**Comment by LukeCEL** [comment]
> Where do I put the license/credit information?

In [README](https://github.com/CelestiaProject/Celestia/blob/master/README) there's a part that says:

>  ```
> # Andrew Tribick
>     Significant update of the star.dat base based on new HIP Reduction of the
>     Raw data, Floor van Leeuwen, 2007.
> ```

You can change that part so it reflects the new data. Also, if you include the celestia-gaia-stardb in its entirety, then the credit will already be there.

**Comment by 375gnu** [comment]
> Where do I put the license/credit information?

In content repo

> Should I move the data-related parts of the readme from Celestia to CelestiaContent and edit it here?

Just copy it from code redo and remove irrelevant parts

**Comment by SevenSpheres** [comment]
> In README there's a part that says:
< snip >
You can change that part so it reflects the new data. Also, if you include the celestia-gaia-stardb in its entirety, then the credit will already be there.

That's what I *would* do, but @375gnu says:

> In content repo
> Just copy it from code redo and remove irrelevant parts

which confuses me...this really doesn't seem like the best way to do things...

**Comment by 375gnu** [comment]
@SevenSpheres that's essentially the same

**Comment by SevenSpheres** [comment]
No, the first is updating the readme, the second is making a copy of it and updating that.

**Comment by 375gnu** [comment]
we dont have a readme in this repo. so how will you change it?

**Comment by SevenSpheres** [comment]
I copied part of the readme here, not sure if it needs more license information than what I have there. I also credited Luke for his nearstars.stc update.

File: README [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        spectbins.stc)
+
+        World-capitals.ssc
+
+    Asterisms.dat
+
+# Andrew Tribick
+    Significant update of the stars.dat base based on Gaia DR2 data
+    (see: https://github.com/ajtribick/celestia-gaia-stardb);
+	licensed under CC-BY-SA 4.0.
```
Comment (375gnu) [inline README:23]: extra space

File: README [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+       CHARM2 stellar radii (charm2.stc)
+
+	   Open cluster catalog (openclusters.dsc)
```
Comment (375gnu) [inline README:27]: extra space

File: README (hunk 1 above)
Comment (SevenSpheres) [reply README:23]: Weird, in Notepad++ I don't see an extra space.

Comment (375gnu) [reply README:23]: because this is TAB actually. and your editor shows it as 4 spaces. use only spaces

**Comment by SevenSpheres** [comment]
@eyvallah @pirogronian

**Comment by SevenSpheres** [comment]
okay to merge now?

**Comment by 375gnu** [comment]
maybe we should test this db on mobile phones?

**Comment by eyvallah** [comment]
@375gnu I'll build an android version for testing

**Comment by SevenSpheres** [comment]
@eyvallah's test build works, but there's a fairly long loading time for the star database.

**Comment by SevenSpheres** [comment]
Would it be possible to use this star database for desktop Celestia, but use the [XHIP database](https://github.com/ajtribick/celestia-xhip) for mobile Celestia?

**Comment by 375gnu** [comment]
On my phone fps is much lower, I can't say how much as it's not shown, but it seems that it's around 10.

@SevenSpheres what's the difference between our current HIP and this XHIP databases?

**Comment by SevenSpheres** [comment]
> what's the difference between our current HIP and this XHIP databases?

Updated data + slightly more stars

**Comment by 375gnu** [comment]
Then it sounds like a good idea for me.

**Comment by SevenSpheres** [comment]
Is this ready to be merged?

**Comment by 375gnu** [comment]
What about this:

> Would it be possible to use this star database for desktop Celestia, but use the [XHIP database](https://github.com/ajtribick/celestia-xhip) for mobile Celestia?

**Comment by ajtribick** [comment]
Regarding using the XHIP-derived database on mobile and the Gaia DR2+XHIP-derived file on desktop, this would result in the XHIP stars being at slightly different positions dependent on platform. You may want to check what that would do to cel URLs. You should also consider what happens to things like the extrasolar planets files, which may or may not depend on stars that are defined in the desktop version but would be missing on mobile.

**Comment by SevenSpheres** [comment]
> What about this:

That's something the devs would need to set up.

> You should also consider what happens to things like the extrasolar planets files, which may or may not depend on stars that are defined in the desktop version but would be missing on mobile.

**Sirius_Alpha**'s exoplanet files include *all* host stars including HIP ones, so that's not a problem.

**Comment by LukeCEL** [comment]
> Regarding using the XHIP-derived database on mobile and the Gaia DR2+XHIP-derived file on desktop, this would result in the XHIP stars being at slightly different positions dependent on platform.

Could this problem be fixed by using taking the Gaia DR2+XHIP stars.dat, but manually cutting out the Tycho-only stars using `startextdump` and `makestardb`?

**Comment by ajtribick** [comment]
> Could this problem be fixed by using taking the Gaia DR2+XHIP stars.dat, but manually cutting out the Tycho-only stars using `startextdump` and `makestardb`?

Assuming that the serialization and deserialization to/from the text representation roundtrips floats correctly, that should work. I haven't actually ever used those applications so can't vouch for them. The stars.dat format is pretty simple so I'd personally just work with that directly, or tweak the generator scripts to filter out non-HIP identifiers.

**Comment by eyvallah** [comment]
> That's something the devs would need to set up.

it would be good if we have another branch for that, so we don't need to mess with the folder structures, etc. we just need to be sure both branches of (desktop and mobile) should be updated

**Comment by 375gnu** [comment]
no need for branches, cmake can be used to select a proper file

**Comment by SevenSpheres** [comment]
Mobile Celestia has been released now (without data updates), so this and #25 can be merged.

**Pull request merged**